### ISSUE TITLE
Implement AKS deployer

### ISSUE THREAD

**Issue by helayoty** [kubernetes-sigs/kubetest2#193]
Would be great to have an AKS deployer available in `kubetest2`. That will help us integrate with [e2e-framework](https://github.com/kubernetes-sigs/e2e-framework) as we were discussing [here](https://github.com/kubernetes-sigs/e2e-framework/issues/131) 

/help

**Comment by BenTheElder** [comment]
FWIW: see discussion in https://github.com/kubernetes-sigs/kubetest2/issues/190#issuecomment-1132299937

TLDR:
The deployers we have now are meant to be relatively simple, zero dependency (only exec) examples.

kubetest2 is expressly designed to enable implementing more deployers out of tree with tighter integration (e.g. using the EKS or AKS or kops libraries) *without* the dependency hell we had in kubetest, and with independent maintainers.

kubetest2 operates on a binary plugin model, so external deployers are pretty much on the same level as anything in-tree.
we just need *something* in tree for developing the project, so we ported some of the most used deployers from the old kubetest

we should probably improve the docs to more clearly emphasize how the project is meant to be extensible out of tree.

also, at the moment, I am the sole active maintainer, and my focus doesn't lie here, so i'm over extended with the currently available features 😞 

kops is here:
https://pkg.go.dev/k8s.io/kops/tests/e2e/kubetest2-kops
https://kops.sigs.k8s.io/contributing/testing/#kubernetes-e2e-testing

**Comment by spiffxp** [comment]
/close
Looks like https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2163 has implemented `kubetest2-aks`

**Comment by BenTheElder** [comment]
we should consider pointing to some of these projects from the README
